### PR TITLE
#122 Provider Organizations API scaffold.

### PR DESCRIPTION
**Pull request by criske**
PR for #122.
Left puzzle to start implementing Github organizations.

### CODE REVIEW

**Comment by criske** [comment]
When this PR passes I'll continue the follow up pdd within the same bounty.
Just wanted to make sure that I had the right aproach.

File: self-api/src/main/java/com/selfxdsd/api/Organization.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     * description etc...
+     * @return Detail
+     */
+    Detail detail();
+
+    /**
+     * Contains extra information about the organization like name,
+     * description etc...
+     */
+    interface Detail {
```
Comment (amihaiemil) [inline self-api/src/main/java/com/selfxdsd/api/Organization.java:42]: @criske Detail is not needed for now. The json() method is enough if we want some specific info about the Organization... we will be interested only in the repos, most likely.

File: self-api/src/main/java/com/selfxdsd/api/Organization.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     * Organization ID.
+     * @return String.
+     */
+    String organizationId();
+
+    /**
+     * Organization repos.
+     * @return OrganizationRepos.
+     */
+    OrganizationRepos repos();
```
Comment (amihaiemil) [inline self-api/src/main/java/com/selfxdsd/api/Organization.java:23]: @criske Can you rename the interface ``OrganiztionRepos`` to simply ``Repos``? ``Repos`` is an interface which represents more repositories, but where these repositories are coming from is not important.

We will most likely have 2 main implementations of Repos: 1 for an Organization's repositories and 1 for a user's private repositories.

**Comment by amihaiemil** [comment]
@criske When making a PR please, always specify the ticket you are solving in the description, so Github links the Issue with the PR automatically -- it's not enough to put it only in the title.

**Comment by amihaiemil** [comment]
@criske looks good, thanks!

**Comment by amihaiemil** [comment]
@rultor merge it

**Pull request merged**